packages/cli/src/util/lockfile.ts

```diff
@@ -11,8 +11,8 @@ export function lockFilepath(filepath: string): void {
       realpath: false,
     });
   } catch (e) {
-    if (isLockfileError(e) && e.code === "ELOCKED") {
-      e.message = `${filepath} is already in use by another process`;
+    if (isLockfileError(e) && (e.code === "ELOCKED" || e.code === "ENOTDIR")) {
+      e.message = `'${filepath}' already in use by another process`;
```

Comment:
I feel like this error message is not descriptive enough for the case of `ENOTDIR`. 

Maybe have a separate error message like "Non-directory lockfile is deprecated. Most likely '${filepath}' is in use by another process running an old version, or orphaned lockfile during the upgrade"

Reply:
There a few cases where we would get error cde `ENOTDIR`
1. A second Lodestar instance is running with version < 1.15.1
2. Another client such as Teku is running
3. Leftover files during upgrade

I do agree your suggested message is better for case 3. but this is very unlikely and the user will have to manually delete .lock files either way, even before 1.15.1.

For cases 1. and 2. I think the current error is better and includes all details a user has to know and for case 3. the error message is better than what we had before.